pkg/util/hostname.go

```diff
@@ -125,6 +125,15 @@ func GetHostname() (string, error) {
 		log.Debug("Unable to get hostname from GCE: ", err)
 	}
 
+	// FQDN
+	if config.Datadog.GetBool("hostname_fqdn") {
+		name, err = getSystemFQDN()
```

Comment:
let's log a debug message when we try to use the fqdn (similar to what we log for other hostname sources).

Let's also log a warning later when:
* the option is disabled, and
* the canonical hostname we'll return comes from `os.Hostname()`, and
* `os.Hostname() != getSystemFQDN()`, and
* `getSystemFQDN()` works on the platform